gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskType.java

```diff
@@ -0,0 +1,247 @@
+    this.diskTypeId = builder.diskTypeId;
+    this.description = builder.description;
+    this.validDiskSize = builder.validDiskSize;
+    this.selfLink = builder.selfLink;
+    this.defaultDiskSizeGb = builder.defaultDiskSizeGb;
+    this.deprecationStatus = builder.deprecationStatus;
+  }
+
+  /**
+   * Returns the creation timestamp in RFC3339 text format.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc3339.txt">RFC3339</a>
+   */
+  public String creationTimestamp() {
+    return creationTimestamp;
+  }
+
+  /**
+   * Returns the disk type's identity.
+   */
+  public DiskTypeId diskTypeId() {
+    return diskTypeId;
+  }
+
+  /**
+   * Returns an unique identifier for the disk type; defined by the service.
+   */
+  public BigInteger id() {
+    return id;
+  }
+
+  /**
+   * Returns a textual description of the disk type.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
+   * Returns an optional textual description of the valid disk size, such as "10GB-10TB".
+   */
+  public String validDiskSize() {
+    return validDiskSize;
+  }
+
+  /**
+   * Returns a service-defined URL for the disk type.
+   */
+  public String selfLink() {
+    return selfLink;
+  }
+
+  /**
+   * Returns the service-defined default disk size in GB.
+   */
+  public Long defaultDiskSizeGb() {
+    return defaultDiskSizeGb;
+  }
+
+  /**
+   * Returns the deprecation status of the disk type. If {@link DeprecationStatus#status()} is
+   * either {@link DeprecationStatus.Status#DELETED} or {@link DeprecationStatus.Status#OBSOLETE}
+   * the disk type should not be used. Returns {@code null} if the disk type is not deprecated.
+   */
+  public DeprecationStatus<DiskTypeId> deprecationStatus() {
+    return deprecationStatus;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("id", id)
+        .add("creationTimestamp", creationTimestamp)
+        .add("description", description)
+        .add("validDiskSize", validDiskSize)
+        .add("selfLink", selfLink)
+        .add("defaultDiskSizeGb", defaultDiskSizeGb)
+        .add("deprecationStatus", deprecationStatus)
+        .toString();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(diskTypeId);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof DiskType && Objects.equals(toPb(), ((DiskType) obj).toPb());
+  }
+
+  com.google.api.services.compute.model.DiskType toPb() {
+    com.google.api.services.compute.model.DiskType diskTypePb =
+        new com.google.api.services.compute.model.DiskType();
+    diskTypePb.setId(id);
+    diskTypePb.setCreationTimestamp(creationTimestamp);
+    diskTypePb.setDescription(description);
+    diskTypePb.setValidDiskSize(validDiskSize);
+    diskTypePb.setSelfLink(selfLink);
+    diskTypePb.setDefaultDiskSizeGb(defaultDiskSizeGb);
+    diskTypePb.setZone(diskTypeId.zoneId().toUrl());
```

Comment:
So that redundancy (selfLink and zone fields) is also part of the underlying API...
Did you check with the service owners if that was intentional and if that may ever change?